app/scenes/Settings/components/WebhookSubscriptionNew.tsx

```diff
@@ -0,0 +1,292 @@
+  margin-right: 0.5rem;
+`;
+
+const FieldSet = styled.fieldset`
+  padding-left: 0;
+  border: none;
+
+  &.disabled {
+    opacity: 50%;
+  }
+`;
+
+const GroupGrid = styled.div`
+  display: grid;
+  grid-template-columns: 1fr 1fr;
+
+  &.mobile {
+    grid-template-columns: 1fr;
+  }
+`;
+
+const GroupWrapper = styled.div`
+  padding-bottom: 2rem;
+
+  &.mobile {
+    padding-bottom: 1rem;
+  }
+`;
+
+const TextFields = styled.div`
+  display: flex;
+  flex-direction: column;
+
+  margin-bottom: 2rem;
+`;
+
+type Props = {
+  onSubmit: () => void;
+};
+
+interface FormData {
+  name: string;
+  url: string;
+  events: string[];
+}
+
+function joinClasses(...classes: (string | boolean | undefined)[]) {
+  return classes.filter(Boolean).join(" ");
+}
+
+function WebhookSubscriptionNew({ onSubmit }: Props) {
+  const { webhookSubscriptions } = useStores();
+  const { showToast } = useToasts();
+  const { t } = useTranslation();
+  const {
+    register,
+    handleSubmit: formHandleSubmit,
+    formState,
+    watch,
+    setValue,
+  } = useForm<FormData>({ mode: "all", defaultValues: { events: [] } });
+
+  const handleSubmit = React.useCallback(
+    async (data: FormData) => {
+      try {
+        const events = Array.isArray(data.events) ? data.events : [data.events];
+
+        const toSend = {
+          ...data,
+          events,
+        };
+
+        await webhookSubscriptions.create(toSend);
+        showToast(
+          t("Webhook subscription created", {
+            type: "success",
+          })
+        );
+        onSubmit();
+      } catch (err) {
+        showToast(err.message, {
+          type: "error",
+        });
+      }
+    },
+    [t, showToast, onSubmit, webhookSubscriptions]
+  );
+
+  const events = watch("events");
+  const selectedGroups = events.filter((e) => !e.includes("."));
+  const isAllEventSelected = events.includes("*");
+  const filteredEvents = events.filter((e) => {
+    const [beforePeriod] = e.split(".");
+
+    return (
+      selectedGroups.length === 0 ||
+      e === beforePeriod ||
+      !selectedGroups.includes(beforePeriod)
+    );
+  });
+
+  const isMobile = useMobile();
+
+  useEffect(() => {
+    if (isAllEventSelected) {
+      setValue("events", ["*"]);
+    }
+  }, [isAllEventSelected, setValue]);
+
+  useEffect(() => {
+    if (!isEqual(events, filteredEvents)) {
+      setValue("events", filteredEvents);
+    }
+  }, [events, filteredEvents, setValue]);
+
+  function EventCheckbox({ label, value }: { label: string; value: string }) {
+    return (
+      <EventCheckboxLabel
+        className={joinClasses(
+          value === "*" && "all-label",
+          value !== "*" &&
+            Object.keys(WEBHOOK_EVENTS).includes(value) &&
+            "group-label"
+        )}
+      >
+        <EventCheckboxText>{label}</EventCheckboxText>
+        <input
+          type="checkbox"
+          defaultValue={value}
+          {...register("events", {})}
+        />
```

Comment:
```suggestion
        <input
          type="checkbox"
          defaultValue={value}
          {...register("events", {})}
        />
        <EventCheckboxText>{label}</EventCheckboxText>
```

Lets flip these around so that the checkboxes are aligned vertically

Reply:
Ahh ya this looks much cleaner this way 👍